### PR TITLE
#85 Fix Zally Swagger Definition

### LINKED ISSUE #85: Current Zally's API spec is incomplete
Description of endpoint `/api-violations` doesn't contain any info on how to actually send swagger spec for linting. 
 
```yaml
paths:
  '/api-violations':
    post:
      summary:
        Retrieves list of API violations based on the uploaded swagger file
      description:
        Retrieves list of API violations based on the uploaded swagger file
      parameters:
        - $ref: '#/parameters/Authorization'
      consumes:
        - application/json
      produces:
        - application/json
      responses:
        200:
          description: API swagger is OK
          schema:
            $ref: '#/definitions/LintingResponse'
        400:
          description: Input file not parsable
          schema:
            $ref: 'https://zalando.github.io/problem/schema.yaml#/Problem'
        default:
          description: An array of error objects
          schema:
            $ref: 'https://zalando.github.io/problem/schema.yaml#/Problem'
      security:
        - oauth2:
          - uid
```

Also description "Retrieves list of API violations based on the uploaded swagger file" sounds very confusing to me - do I really need to upload my swagger file prior to calling this endpoint? 

I propose to add actual spec for request body and change description.

### PR DESCRIPTION
**Pull request by netme**
Closes #85 

📙 Fixes Zally Swagger definition

### Steps:
- [x] Fix issues found by Swagger Editor™
- [x] Fix summary and description
- [x] Check the consistency with existing API
- [x] Try to add an integration test to validate our definition using the latest CLI tool

### CODE REVIEW

File: server/src/main/resources/api/zally-api.yaml [hunk 1]
```diff
@@ -6,17 +6,17 @@ info:
...
+host: "zally.architecture.zalan.do"
 schemes:
   - https
 basePath: /
 securityDefinitions:
   oauth2:
     type: oauth2
     authorizationUrl: https://auth.zalando.com/oauth2/authorize
-    flow: password
+    flow: implicit
```
Comment (fmueller) [inline server/src/main/resources/api/zally-api.yaml]: Sure that it is implicit? In our guidelines we also use password. But I'm not that into OAuth.

**Comment by fmueller** [comment]
Cool that you are working on this :-)

File: server/src/main/resources/api/zally-api.yaml (hunk 1 above)
Comment (netme) [inline server/src/main/resources/api/zally-api.yaml]: I can only say when you are placing `password`, Swagger Editor starts complaining :-)

Comment (fmueller) [inline server/src/main/resources/api/zally-api.yaml]: According to http://zalando.github.io/restful-api-guidelines/security/Security.html we must stick to password flow.

Comment (netme) [inline server/src/main/resources/api/zally-api.yaml]: Thanks Felix for investigations on that. I will change it back then.

**Comment by netme** [comment]
Actually, not bad:

```
$ cli/bin/zally server/src/main/resources/api/zally-api.yaml 

Found the following HINT violations
===================================

Base path can be extracted
	All paths start with prefix '/api-violations'. This prefix could be part of base path.
	http://zalando.github.io/restful-api-guidelines/naming/Naming.html


Summary:
========

MUST violations: 0
SHOULD violations: 0
COULD violations: 0
HINT violations: 1
```

**Comment by netme** [comment]
Adding a small integration test

**Comment by netme** [comment]
Ready for review

File: server/src/main/resources/api/zally-api.yaml [hunk 2]
```diff
@@ -59,6 +66,10 @@ paths:
           description: Input file not parsable
           schema:
             $ref: 'https://zalando.github.io/problem/schema.yaml#/Problem'
+        415:
```
Comment (fmueller) [inline server/src/main/resources/api/zally-api.yaml]: In our guidelines we state that you don't have to document standard http error codes as long as they have default meaning. So, I'd advise to remove Content-Type header and 415 for this reason. Or is Swagger Editor complaining about it?

**Comment by netme** [comment]
@fmueller , I've removed 2 commits :-)

**Comment by fmueller** [comment]
👍

**Pull request merged**